platforms/jvm/plugins-java/src/main/java/org/gradle/jvm/component/internal/DefaultJvmSoftwareComponent.java

```diff
@@ -184,4 +199,70 @@ public void withSourcesJar() {
     public JvmFeatureInternal getMainFeature() {
         return mainFeature;
     }
+
+    @Nullable
+    public JvmTestSuite getTestSuite() {
+        return testSuite;
+    }
+
+    @Override
+    public void consistentResolution(Action<? super JavaResolutionConsistency> action, Project project) {
+        action.execute(new IntraComponentJavaResolutionConsistency(project.getExtensions().getByType(SourceSetContainer.class), project.getConfigurations()));
+    }
+
+    private JvmTestSuite configureBuiltInTest(Project project) {
+        /*
+         * At some point we may want to create a test suite per component, but for now we only want to create one for the `java`
+         * component, in case multiple DefaultJvmSoftwareComponents are created.
+         */
+        TestingExtension testing = project.getExtensions().findByType(TestingExtension.class);
+        if (null != testing && JvmConstants.JAVA_COMPONENT_NAME.equals(getName())) {
```

Comment:
Instead of special-casing the `java` component, what do you think about just accepting the test suite name as a parameter? That way we can keep the constants like `JvmConstants.JAVA_COMPONENT_NAME` and `JavaPluginHelper.DEFAULT_TEST_SUITE_NAME` out of this class -- making it a bit more generic